operator/v1/types_ingress.go

```diff
@@ -556,6 +571,44 @@ const (
 	AWSNetworkLoadBalancer AWSLoadBalancerType = "NLB"
 )
 
+// AWSSubnets contains a list of references to AWS subnets by
+// ID or name.
+// +kubebuilder:validation:XValidation:rule=`has(self.ids) && has(self.names) ? size(self.ids + self.names) <= 10 : true`,message="the total number of subnets cannot exceed 10"
+// +kubebuilder:validation:XValidation:rule=`has(self.ids) || has(self.names)`,message="must specify at least 1 subnet name or id"
+type AWSSubnets struct {
+	// ids specifies a list of AWS subnets by subnet ID.
+	// Subnet IDs must start with "subnet-", consist only of
+	// alphanumeric characters, and be exactly 24 characters
+	// long.
+	// characters long.
+	// +optional
+	// +listType=atomic
+	// +kubebuilder:validation:XValidation:rule=`self.all(x, self.exists_one(y, x == y))`,message="subnet ids cannot contain duplicates"
+	// +kubebuilder:validation:MaxItems=10
+	IDs []AWSSubnetID `json:"ids,omitempty"`
+
+	// names specifies a list of AWS subnets by subnet name.
+	// Subnet names must NOT start with "subnet-".
```

Comment:
Need to explain the maximimal length and that entries must be unique and that they may not contain commas

Reply:
Done.